### ISSUE TITLE
Creating protected classes

### ISSUE THREAD

**Issue by SeanFarrow** [AutoFixture/AutoFixture#547]
Hi,

I am writing a unit test where I need to create an instance of the ParameterInfo class who's constructor is protected. Is there a way to do this with AutoFixture. Or, would Albedo help in this case.
Cheers
Sean.

**Comment by ploeh** [comment]
`ParameterInfo` values are a pain to get at, because you first need a [`MethodBase`](https://msdn.microsoft.com/en-us/library/system.reflection.methodbase) object on which you can invoke [`GetParameters`](https://msdn.microsoft.com/en-us/library/system.reflection.methodbase.getparameters). If there's an easier way, I haven't discovered what it is.

While Albedo can help you with statically-typed access to `MethodInfo` objects, it currently can't give you strongly-typed access to `ParameterInfo` objects. If there's nice, type-safe way to get at these objects, I've yet to figure out what it is.

In Albedo, we use [this test-specific class](https://github.com/ploeh/Albedo/blob/master/Src/Albedo.UnitTests/TypeWithParameters.cs) to create `ParameterInfo` objects for test purposes.

**Comment by ploeh** [comment]
Theoretically, since `ParameterInfo` isn't sealed, you could derive a test-specific class from it and make its constructor public. Even more theoretically, you could use one of AutoFixture's [auto-mocking containers](http://blog.ploeh.dk/2013/03/11/auto-mocking-container) to do this, but IIRC, the APIs exposed by the Reflection classes are too big and complicated, and something always goes wrong that causes me to abandon that approach.

**Comment by ploeh** [comment]
tl;dr: **no.**

Does that answer your question?

**Comment by SeanFarrow** [comment]
Yes, I have taken the same approach as you did with albedo, so this can be closed!
